<launch>

  <!-- defines name of the namespace -->
  <arg name="GS_NS" default="$(optenv GS_NS gs)" />

  <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg GS_NS)_iroc_mission_management_manager" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <group ns="$(arg GS_NS)">

    <!-- IROCMissionManagement nodelet -->
    <node pkg="nodelet" type="nodelet" name="iroc_mission_management" args="$(arg nodelet) iroc_mission_management/IROCMissionManagement $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <param name="config" value="$(find iroc_mission_management)/config/config.yaml" />

      <!-- Subscribers -->
      
      <!-- Services -->
      <remap from="~svc/mission_activation" to="/mission_manager/mission_activation" />
      <remap from="~svc/mission_pausing" to="/mission_manager/mission_pausing" />

      <!-- Services -->
      <remap from="~ac/waypoint_mission" to="/mission_manager" />

    </node>

  </group>

</launch>
